componentapiexample/src/main/java/net/gini/android/vision/component/GiniVisionActivity.java

```diff
@@ -163,14 +166,64 @@ protected void onCreate(Bundle savedInstanceState) {
         configureLogging();
         setupGiniVisionCoordinator();
         if (savedInstanceState == null) {
-            showCamera();
+            final Intent intent = getIntent();
+            if (isIntentActionViewOrSend(intent)) {
+                startGiniVisionLibraryForImportedFile(intent);
+            } else {
+                showCamera();
+            }
         } else {
             initState(savedInstanceState);
             retainFragment();
         }
     }
 
     @Override
+    protected void onNewIntent(final Intent intent) {
+        super.onNewIntent(intent);
+        if (isIntentActionViewOrSend(intent)) {
+            startGiniVisionLibraryForImportedFile(intent);
+        }
+    }
+
+    private void startGiniVisionLibraryForImportedFile(final Intent importedFileIntent) {
+        try {
+            final Document document = GiniVision.createDocumentForImportedFile(importedFileIntent,
+                    this);
+            if (document.isReviewable()) {
+                showFragment(getReviewFragment(document), R.string.title_review);
+            } else {
+                showFragment(getAnalysisFragment(document), R.string.title_review);
+            }
+        } catch (ImportedFileValidationException e) {
+            e.printStackTrace();
+            String message = "File cannot be analyzed";
+            if (e.getValidationError() != null) {
+                switch (e.getValidationError()) {
+                    case TYPE_NOT_SUPPORTED:
+                        message = "File type not supported.";
```

Comment:
This messages should be extracted to a `strings` file and written in german, right?